### PR TITLE
Add security group to subnet

### PR DESCRIPTION
**Pull request by JargoonPard**
@spinnaker/netflix-reviewers @duftler PTAL

### CODE REVIEW

**Comment by ajordens** [comment]
Needs a rebase but otherwise looks fine to me (_only touches Azure packages_).

File: clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/CreateAzureServerGroupAtomicOperation.groovy [hunk 1]
```diff
@@ -109,4 +87,31 @@ class CreateAzureServerGroupAtomicOperation implements AtomicOperation<Map> {
 
     [serverGroups: [(description.region): [name: description.name]]]
   }
+
+  String createSubnet(String resourceGroupName, String virtualNetworkName) {
```
Comment (ajordens) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/CreateAzureServerGroupAtomicOperation.groovy]: This particular logic probably warrants a unit test or two that cover the various edge cases you've added as TODOs.

Comment (JargoonPard) [inline clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/CreateAzureServerGroupAtomicOperation.groovy]: Yes, we actually have tests covering this already.  Good point though, thanks.
